source/common/http/async_client_impl.h

```diff
@@ -93,7 +98,40 @@ class AsyncStreamImpl : public AsyncClient::Stream,
 public:
   AsyncStreamImpl(AsyncClientImpl& parent, AsyncClient::StreamCallbacks& callbacks,
                   const AsyncClient::StreamOptions& options);
-  ~AsyncStreamImpl() override { router_.onDestroy(); }
+  ~AsyncStreamImpl() override {
+    router_.onDestroy();
+    // UpstreamRequest::onCleanup() is guaranteed to reset the high watermark calls.
```

Comment:
sorry I'm not finding onCleanup - do we have the right function name or is it PEBKAC?

Reply:
`UpstreamRequest::cleanUp` perhaps?